src/btree/bt_split.c

```diff
@@ -2274,6 +2274,11 @@ __wt_split_rewrite(WT_SESSION_IMPL *session, WT_REF *ref, WT_MULTI *multi)
     __wt_page_modify_clear(session, page);
     if (!F_ISSET(S2C(session)->cache, WT_CACHE_EVICT_SCRUB) || multi->supd_restore)
         F_SET_ATOMIC_16(page, WT_PAGE_EVICT_NO_PROGRESS);
+
+    /* Set the instantiated disk image to NULL so it is not freed with the old disk image. */
+    if (replace)
```

Comment:
Here it is a bit hacky but I don't have a better idea given the current code structure. For the split code path, we do the same thing in __split_multi_inmem

```
    /*
     * This code re-creates an in-memory page from a disk image, and adds references to any
     * unresolved update chains to the new page. We get here either because an update could not be
     * written when evicting a page, or eviction chose to keep a page in memory.
     *
     * Reconciliation won't create a disk image with entries the running database no longer cares
     * about (at least, not based on the current tests we're performing), ignore the validity
     * window.
     *
     * Steal the disk image and link the page into the passed-in WT_REF to simplify error handling:
     * our caller will not discard the disk image when discarding the original page, and our caller
     * will discard the allocated page on error, when discarding the allocated WT_REF.
     */
    WT_RET(__wt_page_inmem(session, ref, multi->disk_image, WT_PAGE_DISK_ALLOC, &page, &prepare));
    multi->disk_image = NULL;
```
However, we cannot share the same logic with single page rewrite as it uses a fake multi.

Reply:
Instead of freeing it inside the function, how about freeing it in the caller location? That will be much simpler?

Reply:
I can't. The old disk image is discard right after in the same function.

Reply:
line 2277 __wt_ref_out(session, ref);

Reply:
Sorry, I didn't clearly understand. If the page is already freed here what is the need for resetting the `mod_disk_image` to NULL?

Reply:
To not free the disk image. Otherwise with the change __wt_free(session, mod->mod_disk_image); the new disk image will be freed.

Reply:
Ok, understood. How about setting the `mod_disk_image` to NULL once the `multi.disk_image` is set with that value, so there is no need to pass any flag?

Reply:
Ok, understand. But we need to handle the case __wt_split_rewrite fails. We need some variable to hold the pointer temporarily.

Reply:
done